shell/platform/darwin/common/framework/Source/FlutterChannels.mm

```diff
@@ -108,7 +102,7 @@ - (void)setMessageHandler:(FlutterMessageHandler)handler {
     return;
   }
   // Grab reference to avoid retain on self.
-  NSObject<FlutterMessageCodec>* codec = _codec;
+  __weak NSObject<FlutterMessageCodec>* codec = _codec;
```

Comment:
This is wrong, codec needs to be captured by the block to live longer than the channel object.